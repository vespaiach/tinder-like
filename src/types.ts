export interface APIResult<T> {
    ok: boolean;
    body?: {
        data: T[];
        total: number;
        page: number;
    };
    error?: string;
}

export interface UserProfile {
    id: string;
    lastName: string;
    firstName: string;
    email: string;
    title: string;
    picture: string;
}

/**
 * Provider a way to get data from server
 */
export type DataProvider = {
    getList(page: number): Promise<APIResult<UserProfile>>;
};
