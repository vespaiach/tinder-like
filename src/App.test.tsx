import React from 'react';
import { fireEvent, render, getByTestId } from '@testing-library/react';

import AppShell from './App';
import { DataProvider } from './types';

jest.mock('./LikedPage');
jest.mock('./HomePage');

/**
 * Mocking dataProvider
 */
const dataProvider: DataProvider = {
    getList: (p: number) => Promise.resolve({ ok: true }),
};

/**
 * Taking snapshot of app shell
 */
describe('Test app shell', () => {
    test('should be on home page', () => {
        const { container } = render(<AppShell dataProvider={dataProvider} />);
        expect(container).toMatchSnapshot();
    });
    test('should be on liked page', () => {
        const { container } = render(<AppShell dataProvider={dataProvider} />);
        fireEvent.click(getByTestId(container, 'liked-page'));
        expect(container).toMatchSnapshot();
    });
});
