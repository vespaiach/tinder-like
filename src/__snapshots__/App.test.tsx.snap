// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app shell should be on home page 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="makeStyles-navigation-1"
        >
          <button
            aria-label="go to home page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            data-testid="home-page"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <hr
            class="MuiDivider-root MuiDivider-vertical"
          />
          <button
            aria-label="list of liked profiles"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            data-testid="liked-page"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.35 20.13c-.76.69-1.93.69-2.69-.01l-.11-.1C5.3 15.27 1.87 12.16 2 8.28c.06-1.7.93-3.33 2.34-4.29 2.64-1.8 5.9-.96 7.66 1.1 1.76-2.06 5.02-2.91 7.66-1.1 1.41.96 2.28 2.59 2.34 4.29.14 3.88-3.3 6.99-8.55 11.76l-.1.09z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-appShell-2"
        >
          <div>
            home page
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test app shell should be on liked page 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="makeStyles-navigation-3"
        >
          <button
            aria-label="go to home page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            data-testid="home-page"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <hr
            class="MuiDivider-root MuiDivider-vertical"
          />
          <button
            aria-label="list of liked profiles"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            data-testid="liked-page"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.35 20.13c-.76.69-1.93.69-2.69-.01l-.11-.1C5.3 15.27 1.87 12.16 2 8.28c.06-1.7.93-3.33 2.34-4.29 2.64-1.8 5.9-.96 7.66 1.1 1.76-2.06 5.02-2.91 7.66-1.1 1.41.96 2.28 2.59 2.34 4.29.14 3.88-3.3 6.99-8.55 11.76l-.1.09z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-appShell-4"
        >
          <div>
            liked page
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
